### PR TITLE
48 add yearly monthly sub switch

### PR DESCRIPTION
**Pull request by martmull**
close #48
- add cancel subscription link to billing portal
- add change billing from monthly/yearly to yearly/monthly

image
image
image
image

### CODE REVIEW

File: packages/twenty-server/src/engine/core-modules/billing/stripe/stripe.service.ts [hunk 1]
```diff
@@ -105,4 +106,23 @@ export class StripeService {
...
+        stripeSubscriptionItem.stripeSubscriptionItemId,
+        {
+          price: stripePriceId,
+          quantity: stripeSubscriptionItem.quantity,
+        },
+      );
+
+      return true;
+    } catch (e) {
+      return false;
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/billing/stripe/stripe.service.ts:125]: it looks suspicious to return true or false depending on the success of the update. I would let the exception or try catch but trigger a more comprehensive exception in the catch. Then we can catch it outside of the service or let it back to the graphql response

File: packages/twenty-server/src/engine/core-modules/billing/billing.service.ts [hunk 2]
```diff
@@ -154,6 +154,32 @@ export class BillingService {
     return session.url;
   }
 
+  async updateBillingSubscription(user: User): Promise<boolean> {
+    const billingSubscription = await this.getCurrentBillingSubscription({
+      workspaceId: user.defaultWorkspaceId,
+    });
+    const newInterval =
+      billingSubscription?.interval === 'year' ? 'month' : 'year';
+    const billingSubscriptionItem = await this.getBillingSubscriptionItem(
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/billing/billing.service.ts:163]: are all the edge cases covered? what happens if there is no billingSubscription for example?

File: packages/twenty-front/src/pages/settings/SettingsBilling.tsx [hunk 3]
```diff
@@ -26,9 +34,45 @@ const StyledInvisibleChat = styled.div`
...
 `;
 
+type SwitchInfo = {
+  newInterval: string;
+  to: string;
+  from: string;
+  impact: string;
+};
+
+const monthlySwitchInfo: SwitchInfo = {
```
Comment (charlesBochet) [inline packages/twenty-front/src/pages/settings/SettingsBilling.tsx:44]: per project convention constants should be in upper SNAKE_CASE

File: packages/twenty-server/src/engine/core-modules/billing/stripe/stripe.service.ts (hunk 1 above)
Comment (martmull) [reply packages/twenty-server/src/engine/core-modules/billing/stripe/stripe.service.ts]: yes you are right. Updated

**Review by charlesBochet — APPROVED** [review]
LGTM!

**Pull request merged**